model/syncPackage/SyncPackageService.php

```diff
@@ -0,0 +1,187 @@
+{
+    /**
+     *  FileStorageService implements SyncPackageStorageInterface
+     */
+    const OPTION_STORAGE = 'storage';
+    /**
+     *  MigrationService implements MigrationInterface
+     */
+    const OPTION_MIGRATION = 'migration';
+    /**
+     *  Data limit per one package.
+     */
+
+    const PARAM_LTI_USER = 'lti_user';
+    const PARAM_DELIVERY_LOG = 'delivery_log';
+    const PARAM_RESULTS = 'results';
+    const PARAM_TEST_SESSION = 'test_session';
+    const PARAM_LIMIT = 'limit';
+
+    /**
+     * @var SyncPackageStorageInterface
+     */
+    private $storageService;
+
+    /**
+     * @var MigrationInterface
+     */
+    private $migrationService;
+
+    /**
+     * @return mixed|storage\SyncPackageStorageInterface
+     */
+    public function getStorageService()
+    {
+        if (!$this->storageService) {
+            $this->storageService = $this->getOption(self::OPTION_STORAGE);
+            $this->storageService->setServiceLocator($this->getServiceLocator());
+        }
+        return $this->storageService;
+    }
+
+    /**
+     * @return array|SyncQueueInterface
+     */
+    public function getSyncQueueService()
+    {
+        return $this->getServiceLocator()->get(SyncQueueService::SERVICE_ID);
+    }
+
+    /**
+     * @return mixed|SyncPackageDataProviderInterface
+     */
+    public function getDataProviderService()
+    {
+        return $this->getServiceLocator()->get(SyncPackageDataProviderServiceInterface::SERVICE_ID);
+    }
+
+    /**
+     * @return MigrationInterface
+     * @throws SyncClientException
+     */
+    public function getMigrationService()
+    {
+        if (!$this->migrationService) {
+            $hasStorage = true;
+            try {
+                $migrationService = $this->getOption(self::OPTION_MIGRATION);
+                if ($migrationService instanceof MigrationInterface) {
+                    /** @var SyncQueueStorageInterface storageService */
+                    $this->migrationService = $migrationService;
+                    $this->migrationService->setServiceLocator($this->getServiceLocator());
+                } else {
+                    $hasStorage = false;
+                }
+            } catch (\Exception $e) {
+                $hasStorage = false;
+            }
+            if (!$hasStorage) {
+                throw new SyncClientException('taoSyncClient MigrationStorage is not initialized');
+            }
+        }
+        return $this->migrationService;
+    }
+
+    /**
+     * @param array $dataTypes
+     * @param integer $limit
+     * @return common_report_Report
+     * @throws SyncClientException
+     * @throws common_exception_Error
+     */
+    public function create($dataTypes = [], $limit = 0)
+    {
+        $report = common_report_Report::createInfo('Package creation started');
+        if ($this->getStorageService()->isValid()) {
+            $queuedTasks = $this->getSyncQueueService()->getTasks($dataTypes, $limit);
```

Comment:
as discussed with you it will be a possibility to export all data but in separate files (with -e parameter for the script) 
please approve it here, @SashaTAO

Reply:
Yes, I think it should work